src/nvim/lua/treesitter.c

```diff
@@ -992,6 +996,50 @@ static int node_parent(lua_State *L)
   return 1;
 }
 
+static int node_next_sibling(lua_State *L)
+{
+  TSNode node;
+  if (!node_check(L, 1, &node)) {
+    return 0;
+  }
+  TSNode sibling = ts_node_next_sibling(node);
+  push_node(L, sibling, 1);
+  return 1;
+}
+
+static int node_prev_sibling(lua_State *L)
+{
+  TSNode node;
+  if (!node_check(L, 1, &node)) {
```

Comment:
I think you need to do `lua_upvalue_index` here:
```suggestion
  if (!node_check(L, lua_upvalue_index(1), &node)) {
```

Reply:
but why

Reply:
I don't really know, it feels more "robust" for some weird reason in my head.

Reply:
but it is not an upvalue.

Reply:
Indeed, it wasn't clear for me what an upvalue is.

Reply:
I'm new to writing lua C bindings so I'll double check the docs on this.

Reply:
upvalues is for c closures. This is a method, so `self` is passed on stack position 1 (just as first arg).